### PR TITLE
Implemented multiple question types

### PR DESCRIPTION
**Pull request by Ferdinand737**
# Description
Includes all changes from [Queue Refactor PR](https://github.com/ubco-db/office-hours/pull/54)
- Professors/TAs can now create many question types
- Students can select any of the created question types

**Question types temporarily disabled for async question queue.**
- Async questions will be reworked as well but this PR is already big

### CODE REVIEW

**Comment by Ferdinand737** [comment]
Still need to fix insights page and tests

File: packages/server/src/chatbot/chatbotDocument.entity.ts [hunk 1]
```diff
@@ -6,8 +6,7 @@ import {
   ManyToOne,
   PrimaryGeneratedColumn,
 } from 'typeorm';
-import { ChatbotQuestionModel } from './question.entity';
```
Comment (d3li0n) [inline packages/server/src/chatbot/chatbotDocument.entity.ts:9]: I don't think we should remove this.

File: packages/server/test/question.integration.ts [hunk 2]
```diff
@@ -382,8 +448,17 @@ describe('Question Integration', () => {
         userId: user.id,
         courseId: queue.courseId,
       });
+      const questionTypes = [];
+      forEach(QuestionTypes, async (questionType) => {
```
Comment (d3li0n) [inline packages/server/test/question.integration.ts:452]: I don't think foreach works weel with await function, I could be mistaking, but there is an occasional bug where not all items can be added/updated to database. i think we should use for loop?

File: packages/server/migration/1699381717454-fix-organization.ts [hunk 3]
```diff
@@ -1,47 +0,0 @@
-import { MigrationInterface, QueryRunner } from 'typeorm';
-
-export class fixOrganization1699381717454 implements MigrationInterface {
-  name = 'fixOrganization1699381717454';
-
-  public async up(queryRunner: QueryRunner): Promise<void> {
-    await queryRunner.query(
-      `CREATE TABLE "chatbot_interactions_model" ("id" SERIAL NOT NULL, "timestamp" TIMESTAMP NOT NULL, "course" integer, "user" integer, CONSTRAINT "PK_dfa22bd83de6c88afe6caa5ae37" PRIMARY KEY ("id"))`,
```
Comment (d3li0n) [inline packages/server/migration/1699381717454-fix-organization.ts:8]: I don't think we should delete this?

File: packages/common/index.ts [hunk 4]
```diff
@@ -275,15 +275,44 @@ export class Question {
...
-  Clarification = 'Clarification',
-  Testing = 'Testing',
-  Bug = 'Bug',
-  Setup = 'Setup',
-  Other = 'Other',
-}
+export const QuestionTypes: AddQuestionTypeParams[] = [
+  {
+    id: 1,
+    cid: 1,
```
Comment (d3li0n) [inline packages/common/index.ts:281]: Is it default values for question types? What does cid represent?

File: packages/common/index.ts [hunk 5]
```diff
@@ -239,7 +239,7 @@ export type Heatmap = Array<number>
  * @param createdAt - The date string for the time that the Ticket was created. Ex: "2020-09-12T12:00:00-04:00"
  * @param helpedAt - The date string for the time that the TA began helping the Student.
  * @param closedAt - The date string for the time that the TA finished helping the Student.
- * @param questionType - The question type helps distinguish question for TA's and data insights.
+ * @param questionTypes - The question type helps distinguish question for TA's and data insights.
```
Comment (d3li0n) [inline packages/common/index.ts:242]: "The question types help distinguish question for TA's and data insights."? (plural)

File: packages/app/components/AsyncQuestion/Student/AsyncQuestionForm.tsx [hunk 6]
```diff
@@ -1,77 +1,77 @@
-import { ReactElement, useEffect, useState } from "react";
-import Modal from "antd/lib/modal/Modal";
-import { Input, Form, message, Radio } from "antd";
-import styled from "styled-components";
-import { API } from "@koh/api-client";
+import { ReactElement, useEffect, useState } from 'react'
+import Modal from 'antd/lib/modal/Modal'
+import { Input, Form, message, Radio } from 'antd'
```
Comment (d3li0n) [inline packages/app/components/AsyncQuestion/Student/AsyncQuestionForm.tsx:3]: Remove unused imports/variables?

**Review by d3li0n — CHANGES_REQUESTED** [review]
Left comments

File: packages/server/test/question.integration.ts (hunk 2 above)
Comment (Ferdinand737) [reply packages/server/test/question.integration.ts:452]: forEach allows for async, regular for loop does not

File: packages/common/index.ts (hunk 4 above)
Comment (Ferdinand737) [reply packages/common/index.ts:281]: These are just used in tests, cid is course id.

File: packages/server/migration/1699381717454-fix-organization.ts (hunk 3 above)
Comment (Ferdinand737) [reply packages/server/migration/1699381717454-fix-organization.ts:8]: Me neither but i could not run migrations without deleting it.

File: packages/server/src/chatbot/chatbotDocument.entity.ts (hunk 1 above)
Comment (Ferdinand737) [reply packages/server/src/chatbot/chatbotDocument.entity.ts:9]: It was unused

File: packages/common/index.ts (hunk 4 above)
Comment (d3li0n) [reply packages/common/index.ts:281]: If it's only used for tests, do we need it for common?

**Pull request closed without merging**